infrastructure/cdn-in-a-box/traffic_ops/Dockerfile-go

```diff
@@ -20,25 +20,44 @@
 # Based on CentOS 7.2
 ############################################################
 
-FROM centos:7
+# Keep the trafficops-common-deps in Dockerfile-go the same
+# as trafficops-common-deps in Dockerfile to cache the same
+# layer.
+FROM centos:7 as trafficops-common-deps
+
+RUN mkdir /etc/cron.d && \
+    yum -y install https://download.postgresql.org/pub/repos/yum/reporpms/EL-7-x86_64/pgdg-redhat-repo-latest.noarch.rpm && \
+    yum -y install epel-release && \
+    yum -y install      \
+        jq              \
+        bind-utils      \
+        net-tools       \
+        gettext         \
+        perl-JSON-PP    \
+        mkisofs         \
+        isomd5sum       \
+        nmap-ncat       \
+        openssl         \
+        postgresql96 && \
+    yum clean all
+
+FROM    trafficops-common-deps
```

Comment:
> Do these sections need to be duplicated in each Dockerfile? I think we got around that somehow with the cache server images.

The CDN-in-a-Box `mid` and `edge` services use the exact same Docker image and their respective entrypoint scripts are bind-mounted at runtime, so different situation.

The debugging images use the TO and TO Perl images as base images. To generate tags so they can do that, [new services](https://github.com/apache/trafficcontrol/blob/aa03fd37f8/infrastructure/cdn-in-a-box/optional/docker-compose.debugging.yml#L54-L83) are introduced that use the non-debug images and override the command at runtime to exit immediately. The debug services depend on these "non-debug" services to ensure that those tags exist. The reason I went with that ugly hack is that copying entire Dockerfiles just for debugging didn't make sense and would be twice the maintenance work, and I'd prefer to avoid useless "exit-on-start" services like these whenever practical.

One option is to not actually store the Dockerfiles, and instead to generate them on-the-fly using a build system external to Docker like [asciibuild](https://github.com/jbrisbin/asciibuild/blob/master/doc/DOCKER.adoc) ([basho-labs/riak-docker](https://github.com/basho-labs/riak-docker) does this, the Dockerfile is built using [the readme](https://raw.githubusercontent.com/basho-labs/riak-docker/master/README.adoc) as a base).

Compared to the alternatives, keeping a package list in 2 places and instructing maintainers to keep the 2 lists in sync seems relatively tame.

Reply:
Fair enough